src/app/qgisapp.cpp

```diff
@@ -9941,6 +9943,70 @@ void QgisApp::options()
   showOptionsDialog( this );
 }
 
+QMap< QString, QString > QgisApp::projectPropertiesPagesMap()
+{
+  if ( mProjectPropertiesPagesMap.isEmpty() )
+  {
+    QgsProjectProperties *pp = new QgsProjectProperties( mMapCanvas, this );
```

Comment:
This causes a crash on exit, because the QgsProjectProperties is only deleted after the QgisApp destructor is called. And the very last line in that destructor is:
```
  // This function *MUST* be the last one called, as it destroys in
  // particular GDAL. As above objects can hold GDAL/OGR objects, it is not
  // safe destroying them afterwards
  QgsApplication::exitQgis();
  // Do *NOT* add anything here !
```

The solution is to make pp a unique_ptr, so that it's deleted at the end of this block.

Reply:
@nyalldawson 

Fixed